src/main/java/com/northconcepts/datapipeline/examples/amazons3/WriteAnOrcFileToAmazonS3UsingATemporaryFile.java

```diff
@@ -0,0 +1,51 @@
+package com.northconcepts.datapipeline.examples.amazons3;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import com.northconcepts.datapipeline.amazons3.AmazonS3FileSystem;
+import com.northconcepts.datapipeline.core.DataReader;
+import com.northconcepts.datapipeline.csv.CSVReader;
+import com.northconcepts.datapipeline.job.Job;
+import com.northconcepts.datapipeline.orc.OrcDataWriter;
+
+public class WriteAnOrcFileToAmazonS3UsingATemporaryFile {
+
+    private static final String ACCESS_KEY = "YOUR ACCESS KEY";
+    private static final String SECRET_KEY = "YOUR SECRET KEY";
+
+    public static void main(String[] args) throws Throwable {
+        AmazonS3FileSystem s3 = new AmazonS3FileSystem();
+        s3.setBasicAWSCredentials(ACCESS_KEY, SECRET_KEY);
+        s3.open();
+
+        try {
+            File orcFile = File.createTempFile("credit-balance", ".orc");
+            orcFile.deleteOnExit();
+
+            try {
+                DataReader reader = new CSVReader(new File("example/data/input/credit-balance.csv"))
+                        .setFieldNamesInFirstRow(true);
+                OrcDataWriter writer = new OrcDataWriter(orcFile);
+
+                Job.run(reader, writer);
```

Comment:
Similar to above, can we do this before opening the S3 connection?

Reply:
I have refactored code which is better to understand.